editor/plugins/color_channel_selector.cpp

```diff
@@ -0,0 +1,131 @@
+#include "scene/resources/style_box_flat.h"
+
+ColorChannelSelector::ColorChannelSelector() :
+		buttons() {
+	HBoxContainer *container = memnew(HBoxContainer);
+	container->add_theme_constant_override("separation", 0);
+
+	create_button(0, "R", container);
+	create_button(1, "G", container);
+	create_button(2, "B", container);
+	create_button(3, "A", container);
+
+	add_child(container);
+}
+
+void ColorChannelSelector::_notification(int p_what) {
+	if (p_what == NOTIFICATION_THEME_CHANGED) {
+		// It is a common pattern in the editor to use `NOTIFICATION_THEME_CHANGED` to setup theme overrides that need
+		// access to the editor's theme. Unfortunately, in our case, this leads to infinite recursion.
+		if (_setting_up_theme) {
+			return;
+		}
+		_setting_up_theme = true;
+		// PanelContainer's background is invisible in the editor. We need a background.
+		// And we need this in turn because buttons don't look good without background (for example, hover is transparent).
+		Ref<StyleBox> bg_style = get_theme_stylebox("panel", "TabContainer");
+		ERR_FAIL_COND(bg_style.is_null());
+		bg_style = bg_style->duplicate();
+		// The default content margin makes the widget become a bit too large. It should be like mini-toolbar.
+		const float editor_scale = EditorScale::get_scale();
+		bg_style->set_content_margin(SIDE_LEFT, 1.f * editor_scale);
+		bg_style->set_content_margin(SIDE_RIGHT, 1.f * editor_scale);
+		bg_style->set_content_margin(SIDE_TOP, 1.f * editor_scale);
+		bg_style->set_content_margin(SIDE_BOTTOM, 1.f * editor_scale);
+		add_theme_style_override("panel", bg_style);
+		_setting_up_theme = false;
+	}
+}
+
+void ColorChannelSelector::set_available_channels_mask(const uint32_t mask) {
+	for (unsigned int i = 0; i < CHANNEL_COUNT; ++i) {
+		const bool available = (mask & (1 << i)) != 0;
+		Button *button = buttons[i];
+		button->set_visible(available);
+	}
+}
+
+void ColorChannelSelector::on_button_toggled(const bool unused_pressed) {
+	emit_signal("selected_channels_changed");
+}
+
+uint32_t ColorChannelSelector::get_selected_channels_mask() const {
+	uint32_t mask = 0;
+	for (unsigned int i = 0; i < CHANNEL_COUNT; ++i) {
+		Button *button = buttons[i];
+		if (button->is_visible() && buttons[i]->is_pressed()) {
+			mask |= (1 << i);
+		}
+	}
+	return mask;
+}
+
+// Helper
+Vector4 ColorChannelSelector::get_selected_channel_factors() const {
+	Vector4 channel_factors;
+	const uint32_t mask = get_selected_channels_mask();
+	for (unsigned int i = 0; i < 4; ++i) {
+		if ((mask & (1 << i)) != 0) {
+			channel_factors[i] = 1;
+		}
+	}
+	return channel_factors;
+}
+
+void ColorChannelSelector::create_button(const unsigned int p_channel_index, const String p_text, Control *p_parent) {
+	CRASH_COND(p_channel_index >= CHANNEL_COUNT);
+	CRASH_COND(buttons[p_channel_index] != nullptr);
+	Button *button = memnew(Button);
+	button->set_text(p_text);
+	button->set_toggle_mode(true);
+	button->set_pressed(true);
+	// button->set_flat(true);
```

Comment:
Remove commented out code:

```suggestion
	button->set_pressed(true);
```